src/Build/Instance/ProjectTaskInstance.cs

```diff
@@ -141,10 +141,10 @@ string msbuildArchitecture
             new CopyOnWriteDictionary<string, (string, ElementLocation)>(8, StringComparer.OrdinalIgnoreCase),
             new List<ProjectTaskInstanceChild>(),
             location,
-            condition == string.Empty ? null : ElementLocation.EmptyLocation,
-            continueOnError == string.Empty ? null : ElementLocation.EmptyLocation,
-            msbuildRuntime == string.Empty ? null : ElementLocation.EmptyLocation,
-            msbuildArchitecture == string.Empty ? null : ElementLocation.EmptyLocation)
+            string.IsNullOrEmpty(condition) ? null : ElementLocation.EmptyLocation,
```

Comment:
This one technically changes the code path. If `null` were passed before, `ElementLocation.EmptyLocation` would be used. If `null` were passed with this change, it would pass null into the base constructor. I'm not sure what that realistically means for this code though.

Looks like these values would never be null either.
```
        internal ProjectTaskInstance AddTask(string taskName, string condition, string continueOnError, string msbuildRuntime, string msbuildArchitecture)
        {
            ErrorUtilities.VerifyThrowInternalLength(taskName, "taskName");
            ProjectTaskInstance task = new ProjectTaskInstance(taskName, _location, condition ?? String.Empty, continueOnError ?? String.Empty, msbuildRuntime ?? String.Empty, msbuildArchitecture ?? String.Empty);
            this.AddProjectTargetInstanceChild(task);
            return task;
        }
```

Reply:
What would you like to do? I can revert the change if you would like.

Reply:
I lean toward reverting this particular change. In the PR review meeting tomorrow we'll determine the direction we want to go here. Thanks for the quick responses by the way!